src/main/java/org/tikv/common/util/ChannelFactory.java

```diff
@@ -188,22 +199,39 @@ public ChannelFactory(
       int keepaliveTime,
       int keepaliveTimeout,
       int idleTimeout,
+      long connRecycleTime,
+      long certReloadInterval,
       String trustCertCollectionFilePath,
       String keyCertChainFilePath,
       String keyFilePath) {
     this.maxFrameSize = maxFrameSize;
     this.keepaliveTime = keepaliveTime;
     this.keepaliveTimeout = keepaliveTimeout;
     this.idleTimeout = idleTimeout;
+    this.connRecycleTime = connRecycleTime;
     this.certContext =
         new OpenSslContext(trustCertCollectionFilePath, keyCertChainFilePath, keyFilePath);
+
+    File trustCert = new File(trustCertCollectionFilePath);
+    File keyCert = new File(keyCertChainFilePath);
+    File key = new File(keyFilePath);
+
+    if (certReloadInterval > 0) {
+      onCertChange();
+      this.certWatcher =
+          new CertWatcher(1, ImmutableList.of(trustCert, keyCert, key), this::onCertChange);
```

Comment:
why we set interval as 1 in openssl

Reply:
It's a miss typing, change to `certReloadInterval`.